x-pack/plugins/observability_ai_assistant/public/hooks/use_timeline.ts

```diff
@@ -100,12 +98,15 @@ export function useTimeline({
 
   const [isFunctionLoading, setIsFunctionLoading] = useState(false);
 
-  const prevConversationId = usePrevious(conversationId);
+  const prevMessagesLength = usePrevious(messages.length);
```

Comment:
Is `if (messages.length === 1 && prevMessagesLength && prevMessagesLength > 1) {` not solid enough to determine the conversation was reset?

Reply:
I don't know, and that's part of it - it's hard to grok in what scenarios this does and does not make sense. Additionally, what happens if you switch from a conversation that has the same amount of messages?

Reply:
If the amount of messages is `1`, it means we have the initial system prompt and nothing else. So clicking `New chat` won't do anything as you're already on `/new`.

Reply:
In general I'm not loving how `useTimeline` depends on `useConversation` (specifically the `messages` prop.)  Makes it hard for `useTimeline` to support something like a `reset` function, because it is getting messages from elsewhere but tacking on loading / error states inside `useTimeline`.

This was the least invasive method that I could think of that would work.

Reply:
> Makes it hard for useTimeline to support something like a reset function, because it is getting messages from elsewhere but tacking on loading / error states inside useTimeline.

I'm not sure if I follow this specifically, but I agree that there's not a clear boundary of what useTimeline does and what should be left to consumers, so definitely open to shuffling things around here. That being said, I don't think checking the amount of messages is a reliable way to do this. Are you open to moving this out into a separate PR so we can merge the other fixes here?